mbed_host_tests/host_tests_plugins/host_test_plugins.py

```diff
@@ -188,16 +190,18 @@ def check_serial_port_ready(self, serial_port, target_id=None, timeout=60):
             for i in range(timeout):
                 # mbed_lstools.create() should be done inside the loop. Otherwise it will loop on same data.
                 mbeds = mbed_lstools.create()
-                mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
-                if target_id in mbeds_by_tid:
-                    if 'serial_port' in mbeds_by_tid[target_id]:
-                        if mbeds_by_tid[target_id]['serial_port']:
-                            # Only assign if serial port is known (not None)
-                            new_serial_port = mbeds_by_tid[target_id]['serial_port']
-                            if new_serial_port != serial_port:
-                                # Serial port changed, update to new serial port from mbed-ls
-                                self.print_plugin_info("Serial port for tid='%s' changed from '%s' to '%s'..." % (target_id, serial_port, new_serial_port))
-                            break
+                mbed_list = mbeds.list_mbeds(unique_names=True, read_details_txt=True) #list of mbeds present
```

Comment:
I don't think that you need the `read_details_text` and could use the default of False here. You're not using the `platform_unique_id`, so you can also drop `unique_names`.